backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -548,46 +550,50 @@ public void validateDiffMergeRequest(
     CommitEntity internalCommitB = null;
     CommitEntity parentCommit;
     Commit parentCommitProto;
+    String branchOrCommitA = null;
+    String branchOrCommitB = null;
     try (Session readSession = ModelDBHibernateUtil.getSessionFactory().openSession()) {
       RepositoryEntity repositoryEntity =
           repositoryDAO.getRepositoryById(readSession, request.getRepositoryId());
 
-      String errorNameA = null;
-      String errorNameB = null;
       if (!request.getBranchA().isEmpty()) {
-        errorNameA = request.getBranchA();
+        LOGGER.debug("Branch A found in request");
+        branchOrCommitA = request.getBranchA();
         BranchEntity branchAEntity =
             repositoryDAO.getBranchEntity(
                 readSession, repositoryEntity.getId(), request.getBranchA());
         internalCommitA = readSession.get(CommitEntity.class, branchAEntity.getCommit_hash());
       }
 
       if (!request.getBranchB().isEmpty()) {
-        errorNameB = request.getBranchB();
+        LOGGER.debug("Branch B found in request");
+        branchOrCommitB = request.getBranchB();
         BranchEntity branchBEntity =
             repositoryDAO.getBranchEntity(
                 readSession, repositoryEntity.getId(), request.getBranchB());
         internalCommitB = readSession.get(CommitEntity.class, branchBEntity.getCommit_hash());
       }
 
       if (!request.getCommitShaA().isEmpty()) {
-        errorNameA = request.getCommitShaA();
+        LOGGER.debug("Commit A found in request");
+        branchOrCommitA = request.getCommitShaA();
```

Comment:
you can do the substring here, and on the other commit

Reply:
Fixed